lib/rubocop/cop/rails/read_write_attribute.rb

```diff
@@ -18,6 +18,41 @@ module Rails
       class ReadWriteAttribute < Cop
         MSG = 'Prefer `%s` over `%s`.'
 
+        def autocorrect(node)
+          _receiver, method_name, _body = *node
+
+          case method_name
+          when :read_attribute
+            autocorrect_read(node)
+          when :write_attribute
+            autocorrect_write(node)
+          end
+        end
+
+        def autocorrect_read(node)
+          _receiver, _method_name, body = *node
+
+          selfing = "self[#{body.loc.expression.source}]"
+
+          @corrections << lambda do |corrector|
+            corrector.replace(node.loc.expression, selfing)
+          end
+        end
+
+        def autocorrect_write(node)
+          _receiver, _method_name, *args = *node
+          name, value = *args
+
+          name_source = name.loc.expression.source
+          value_source = value.loc.expression.source
+
+          selfing = "self[#{name_source}] = #{value_source}"
+
+          @corrections << lambda do |corrector|
```

Comment:
Move these 3 duplicated lines into the `autocorrect` method and return the "selfing" value. After that change it would be better to name the methods, for example, `..._attribute_replacement`.